src/maintainer/conda_forge_yml.rst

```diff
@@ -291,6 +291,15 @@ modified.  Tools like conda-smithy may modify this, as need.  It has a single
       secure:
         BINSTAR_TOKEN: <some big hash>
 
+upload_on_branch
+----------------
+This parameter restricts uploading access on work from certain branches of the same repo.
+Set the value to the name of the branch that should be granted uploading privileges.
+
+.. code-block:: yaml
+
+    upload_on_branch: master
```

Comment:
Should we give an example with two branch names?

Reply:
Right now it's implemented as a single value, not a list. We tried to come up with an example (https://github.com/conda-forge/conda-smithy/pull/1108#issuecomment-512596349), but didn't have a representative repo to show this. Do you have an idea how it can be used?

Reply:
The matplotlib example you found is the common use case. I personally would lean towards listing all branches as a list in a shared conda_forge_yml than having to modify it for each branch. But no strong opinion from my side. cc @scopatz

Reply:
@croth1 Thank you for your suggestion. I have made some changes.

Reply:
I think an example with multiple banches is needed, even if it is kind of just made up.  It doesn't need to be real.

Reply:
@croth1, how can we have a shared `conda-forge.yml`?

@scopatz, why does it need to have multiple branches? Maintaining a synchronized list is hard and say we make a new branch for numpy v17.x, why should the conda-forge.yml be updated in all the other branches?

Reply:
@isuruf, I agree that maintaining a synchronised list on all branches is messy and technically it is probably not possible to have only one central shared one. Please disregard my comment here.